modules/core/migrations/v7/genesis_test.go

```diff
@@ -0,0 +1,156 @@
+package v7_test
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/stretchr/testify/suite"
+
+	ibcclient "github.com/cosmos/ibc-go/v6/modules/core/02-client"
+	clientv7 "github.com/cosmos/ibc-go/v6/modules/core/02-client/migrations/v7"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	v7 "github.com/cosmos/ibc-go/v6/modules/core/migrations/v7"
+	"github.com/cosmos/ibc-go/v6/modules/core/types"
+	ibctesting "github.com/cosmos/ibc-go/v6/testing"
+)
+
+type MigrationsV7TestSuite struct {
+	suite.Suite
+
+	coordinator *ibctesting.Coordinator
+
+	// testing chains used for convenience and readability
+	chainA *ibctesting.TestChain
+	chainB *ibctesting.TestChain
+}
+
+// TestMigrationsV7TestSuite runs all the tests within this package.
+func TestMigrationsV7TestSuite(t *testing.T) {
+	suite.Run(t, new(MigrationsV7TestSuite))
+}
+
+// SetupTest creates a coordinator with 2 test chains.
+func (suite *MigrationsV7TestSuite) SetupTest() {
+	suite.coordinator = ibctesting.NewCoordinator(suite.T(), 2)
+	suite.chainA = suite.coordinator.GetChain(ibctesting.GetChainID(1))
+	suite.chainB = suite.coordinator.GetChain(ibctesting.GetChainID(2))
+}
+
+// NOTE: this test is mainly copied from 02-client/migrations/v7/genesis_test.go
+func (suite *MigrationsV7TestSuite) TestMigrateGenesisSolomachine() {
+	// create tendermint clients
+	for i := 0; i < 3; i++ {
+		path := ibctesting.NewPath(suite.chainA, suite.chainB)
+
+		suite.coordinator.SetupClients(path)
+
+		err := path.EndpointA.UpdateClient()
+		suite.Require().NoError(err)
+		err = path.EndpointA.UpdateClient()
```

Comment:
repeated codeblock here?

Reply:
it was intentional to update twice to generate a little more state. I can add a comment